internal/resources/cluster/nodepools/resource_node_pool.go

```diff
@@ -482,8 +482,8 @@ func resourceNodePoolCreate(ctx context.Context, d *schema.ResourceData, m inter
 		},
 	}
 
-	if nodePoolRequest.Nodepool.Spec.TkgServiceVsphere == nil {
-		return diag.FromErr(fmt.Errorf("TKGs vsphere nodepool spec has to be provided"))
+	if nodePoolRequest.Nodepool.Spec.TkgServiceVsphere == nil && nodePoolRequest.Nodepool.Spec.TkgVsphere == nil && nodePoolRequest.Nodepool.Spec.TkgAws == nil {
+		return diag.FromErr(fmt.Errorf("nodepool spec has to be provided"))
```

Comment:
@ramya-bangera , 
is this check needed? 
I believe the construct function above is handling corner cases, just validate it once. This check is needed to avoid runtime nil panic. so check on these front.

Even if the check is needed, can you improve the error message because need the flavor block to be populated not just the spec

Reply:
Check is needed and improved the error message as suggested.